### DISCUSSION TITLE
Extend the tracing feature with custom traces

### DISCUSSION THREAD

**Discussion by chbiel** [concourse/concourse#7007] [Ideas]
We recently started playing around with the tracing feature in combination with Jaeger.

I feel that the whole feature is in an early state, as I guess there would be much more to show in the traces.

I would love to see some way (but I actually don't have an idea how), to add my own executions to the traces.
E.g. when I have a task that does following things (simple example):
- npm install
- npm test
- npm build
I would really love to be able to add every of those steps as a separate trace into Jaeger, so I can investigate better, how to improve my build.

I know that this would be possible by splitting those 3 executions into different tasks, but that would also involve streaming of artifacts etc. what in the end would slow the execution down.

I would also like to help implementing it, but I don't have enough experience with tracing/Jaeger/concourse internals to do it or even understand if that is possible at all.

What are your thoughts? Do you thinks it's valuable or even possible?

**Comment by aoldershaw** [comment]
It's currently (technically) possible to emit custom traces in `task`s (and `get`s, `put`s, `check`s) - when tracing is configured we add the `TRACEPARENT` environment variable to each container, which can be used to link traces to the parent span. However, these containers have to emit the spans themselves, and therefore must know e.g. how to speak to Jaeger.

See #5653 for more context, and https://github.com/concourse/concourse/blob/199acfad85499ed82d4c3561687b60ffd6329a73/hack/tracing-task.yml for an example task that makes use of this information to emit spans.

**Reply by chbiel** [reply]
thanks! i will have a look into it and try it out :)
